server/webdriver/tests/standalone/sanity.py

```diff
@@ -0,0 +1,338 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+from datetime import datetime
+from enum import StrEnum
+import os
+
+from absl import app
+from absl import flags
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+from webdriver_manager.chrome import ChromeDriverManager
+
+FLAGS = flags.FLAGS
+
+OUTPUT_DIR = "output"
+
+
+class Mode:
+  HOME = "home"
+  EXPLORE_LANDING = "explore_landing"
+  EXPLORE = "explore"
+
+
+flags.DEFINE_string(
+    "mode",
+    Mode.HOME,
+    f"Specify one of the following modes: {Mode.HOME} (default), {Mode.EXPLORE_LANDING}, {Mode.EXPLORE}",
+)
+
+flags.DEFINE_string("url", None, "URL to start testing from.", required=True)
+
+
+class PageType(StrEnum):
+  UNKNOWN = "Unknown"
+  HOME = "Home"
+  EXPLORE_LANDING = "Explore Landing"
+  EXPLORE = "Explore"
+
+
+class WebPage:
+
+  def __init__(self, page_type: PageType, title: str, url: str) -> None:
+    self.page_type = page_type
+    self.title = title
+    self.url = url
+
+
+class Result:
+
+  def __init__(self, page: WebPage, status: str, comments: str = "") -> None:
+    self.page_type = page.page_type
+    self.title = page.title
+    self.status = status
+    self.url = page.url
+    self.comments = comments
+
+  def to_csv_row(self) -> dict:
+    return {key: str(value) for key, value in self.__dict__.items()}
+
+  def __str__(self) -> str:
+    return ", ".join(self.to_csv_row().values())
+
+
+class WebsiteSanityTest:
+
+  def __init__(self, results_csv_file_path) -> None:
+    self.results_csv_file_path = results_csv_file_path
+    self.results = []
+
+  def __enter__(self):
+    self.driver = webdriver.Chrome(
+        service=Service(ChromeDriverManager().install()))
+    self.file = open(self.results_csv_file_path, "w", newline="")
+    print("Writing results to", self.results_csv_file_path)
+    self.csv_writer = csv.DictWriter(self.file,
+                                     fieldnames=result_csv_columns(),
+                                     lineterminator="\n")
+    self.csv_writer.writeheader()
+    return self
+
+  def __exit__(self, exc_type, exc_value, traceback):
+    self.file.close()
+    print("\nResults written to", self.results_csv_file_path)
+    self.driver.quit()
+
+  def add_result(self, result: Result) -> None:
+    print(f"#{len(self.results) + 1}", result)
+    self.csv_writer.writerow(result.__dict__)
+    self.file.flush()
+    self.results.append(result)
+
+  def home(self, page: WebPage):
+    self.driver.get(page.url)
+
+    page.title = self.driver.title if page.title is None else page.title
+
+    # topic cards
+    topic_cards = find_elems(self.driver, By.CLASS_NAME, "topic-card")
```

Comment:
I would suggest to have a few functions to hold the check/grab components logic, so it's easy to update if the css changes.

Reply:
I tried it but it didn't make it any easier since the DOM aspects are too tightly coupled. Let's discuss if you have ideas and can do a follow-on.

Reply:
SG, we can discuss follow up changes.